scripts/abins/instruments/instrument.py

```diff
@@ -19,14 +23,104 @@ class Instrument:
     def __init__(self, setting: str = ''):
         setting = self._check_setting(setting)
         self._setting = setting
+        self._setting_parameters = None
 
     def get_angles(self):
         """Get a list of detector angles to sample"""
         raise NotImplementedError()
 
-    def get_setting(self):
+    def get_parameter(self, key: str, default: Any = None) -> Any:
+        """Get a named instrument parameter from abins.parameters
+
+        Setting parameters will be checked first and take priority over
+        higher-level instrument parameters.
+
+        e.g. if abins.parameters includes::
+
+            instruments = {
+                'THIS_INSTRUMENT': {'param': 'value-1',
+                                    'settings': {'SETTING':
+                                                 {'param': 'value-2'}}}
+            }
+
+        Then self.get_parameter('param') will return 'value-1' unless the
+        setting is 'SETTING', in which case it will return 'value-2'.
+
+        Args:
+            key: key to be searched in instrument parameters
+            default: Value to be returned if key is not found.
+        """
+        setting_parameters = self.get_setting_parameters()
+        if key in setting_parameters:
```

Comment:
is this supposed to give priority to the value in settings?

Reply:
Yes, settings can overrule non-setting-specific parameters. I'm not sure it makes any difference in practice as they are (so far) never specified in both places; it's just an efficient way of checking both places and skipping the second check if something was already found.

It makes more sense for the setting parameters to overrule the generic ones than vice versa if the situation _does_ arise. (Say one instrument has a whole bunch of settings, and only one modifies the resolution parameters; then the other settings could fall back to an instrument-wide value.)